File: TShock.Plugin.ProjectTemplates/content/Plugin-FSharp/RegexPluginClass.fs
namespace Plugin.Namespace

open System
open Terraria
open TerrariaApi.Server

[<ApiVersion(2, 1)>]
 type Plugin.ClassName = 
    inherit TerrariaPlugin
    new (game) = { inherit TerrariaPlugin(game) }
    override _.Name = "Plugin.Namespace"
    override _.Author = base.Author
    override _.Description = base.Description
    override _.Version = new Version(1, 0)
        
    override _.Initialize() = ()

    override _.Dispose(disposing) = 
        if disposing then ()
        ()
            

File: TShock.Plugin.ProjectTemplates/content/Plugin-CSharp/RegexPluginClass.cs
using Terraria;
using TerrariaApi.Server;

namespace Plugin.Namespace;

[ApiVersion(2, 1)]
public class Plugin.ClassName : TerrariaPlugin
{
    public override string Name => "Plugin.Namespace";
    public override string Author => "Plugin.Author";
    public override string Description => "Plugin.Description";
    public override Version Version => new Version(1, 0);
    public Plugin.ClassName(Main game) : base(game)
    {
    }

    public override void Initialize()
    {
        
    }

    protected override void Dispose(bool disposing)
    {
        if (disposing)
        {

        }
    }
}
